completion/available/makefile.completion.bash

```diff
@@ -1,3 +1,36 @@
-# Add completion for Makefile
-# see http://stackoverflow.com/a/38415982/1472048
-complete -W "\`grep -oE '^[a-zA-Z0-9_-]+:([^=]|$)' Makefile | sed 's/[^a-zA-Z0-9_-]*$//'\`" make
+#!/usr/bin/env bash
+# Bash completion for Makefile
+
+# Loosely adapted from http://stackoverflow.com/a/38415982/1472048
+
+_makecomplete() {
+  COMPREPLY=()
+
+  # https://www.gnu.org/software/make/manual/html_node/Makefile-Names.html
+  local files=()
+  while IFS='' read -r line; do
+    files+=("$line")
+  done < <(find . -maxdepth 1 -regextype posix-extended -regex '.*(GNU)?[Mm]akefile$' -printf '%f\n')
```

Comment:
I tested this on my mac with the following file named `GNUmakefile`:
```makefile
foo:
        frobnicate > foo

%: force
        @$(MAKE) -f Makefile $@
force: ;
```

```
$ make --version
GNU Make 3.81 ...

$ make fo     #hitting tab here
find: -regextype: unknown primary or operator
```
A [quick internet search](https://ponderthebits.com/2017/01/know-your-tools-linux-gnu-vs-mac-bsd-command-line-utilities-grep-strings-sed-and-find/) yields some interesting results. Appears that there are some core differences with BSD linux tools and GNU.

Instead of `--regextype` using `-E`:
```
$ find -E . -maxdepth 1 -regex '.*(GNU)?[Mm]akefile$' -printf '%f\n'
find: -printf: unknown primary or operator
```
But, looks like `printf` is not necessary in bsd find:
```
$ find -E . -maxdepth 1 -regex '.*(GNU)?[Mm]akefile$'
./GNUmakefile
```
Replacing the find command in 13 with the bsd `find` command above yields correct results:
```
$ make fo    # hitting tab here
foo    force  
```

Reply:
@jpmcb Yeah, this is exactly why I wanted to test on mac. Sorry you beat me to the rabbit hole :p.

I kinda expected this. Unfortunately running the command you came up with errors on linux for me:
```bash
$ make find: unknown predicate `-E'
```

I'm just going to hardcode the 3 _well known_ names that make looks for, and test for their existence. that will keep more of this to bash and reduce the concerns of portability with `find`.